paasta_tools/paasta_metastatus.py

```diff
@@ -84,13 +87,17 @@ def filter_mesos_state_metrics(dictionary):
 
 def get_extra_mesos_slave_data(mesos_state):
     slaves = dict((slave['id'], {
-        'free_resources': Counter(filter_mesos_state_metrics(slave['resources'])),
+        'total_resources': Counter(filter_mesos_state_metrics(slave['resources'])),
         'hostname': slave['hostname'],
+        'free_resources': Counter(filter_mesos_state_metrics(slave['resources'])),
     }) for slave in mesos_state['slaves'])
 
     for framework in mesos_state.get('frameworks', []):
         for task in framework.get('tasks', []):
-            slaves[task['slave_id']]['free_resources'].subtract(filter_mesos_state_metrics(task['resources'])),
+            mesos_metrics = filter_mesos_state_metrics(task['resources'])
+            for resource, count in mesos_metrics.iteritems():
+                slaves[task['slave_id']]['free_resources'][resource] = slaves[
+                    task['slave_id']]['total_resources'][resource] - count
```

Comment:
This returns the difference between `total_resources` and the `free_resource` of only the last task in `mesos_metrics.iteritems()`. What's wrong with iterating through and using `Counter.subtract()` in this case?

Example of this failing in norcal-devc:
```
>>> for line in paasta_metastatus.assert_extra_slave_data(get_mesos_state_from_leader()):
...     print line
    Hostname                                    CPU (free/total)  RAM (free/total)     Disk (free/total)
    srv4-uswest1cdevc.dev.yelpcorp.com          5.00/7.00         59141.00/59441.00    130501.00/131525.00
    srv4-uswest1adevc.dev.yelpcorp.com          6.90/7.00         55441.00/59441.00    130500.00/131524.00
    srv2-uswest1adevc.dev.yelpcorp.com          11.90/12.00       28150.00/28650.00    289291.00/290315.00
    srv1-uswest1cdevc.dev.yelpcorp.com          11.90/12.00       28341.00/28841.00    289291.00/290315.00
    srv3-uswest1adevc.dev.yelpcorp.com          12.90/13.00       26363.00/26863.00    289291.00/290315.00
    srv3-uswest1cdevc.dev.yelpcorp.com          12.90/13.00       28208.00/28708.00    289291.00/290315.00
    srv2-uswest1cdevc.dev.yelpcorp.com          12.90/13.00       28480.00/28980.00    289291.00/290315.00
    srv1-uswest1adevc.dev.yelpcorp.com          13.00/14.00       19030.00/29030.00    289291.00/290315.00
    10-40-10-100-uswest1adevc.dev.yelpcorp.com  38.00/39.00       146187.00/159187.00  279216.00/280240.00
    10-40-10-243-uswest1adevc.dev.yelpcorp.com  38.67/39.00       158487.00/159187.00  279216.00/280240.00
    10-40-10-198-uswest1adevc.dev.yelpcorp.com  38.90/39.00       158687.00/159187.00  279216.00/280240.00
    10-40-10-234-uswest1adevc.dev.yelpcorp.com  38.90/39.00       158687.00/159187.00  279216.00/280240.00
    10-40-10-35-uswest1adevc.dev.yelpcorp.com   38.90/39.00       158687.00/159187.00  279216.00/280240.00
    10-40-11-114-uswest1adevc.dev.yelpcorp.com  38.90/39.00       158687.00/159187.00  279216.00/280240.00
    10-40-11-167-uswest1adevc.dev.yelpcorp.com  38.90/39.00       158887.00/159187.00  279216.00/280240.00
    10-40-10-163-uswest1adevc.dev.yelpcorp.com  38.90/39.00       158987.00/159187.00  279216.00/280240.00
    10-40-10-80-uswest1adevc.dev.yelpcorp.com   38.90/39.00       158987.00/159187.00  279216.00/280240.00
```
VS
```
paasta metastatus -c norcal-devc -vvv
.....
    Hostname                                    CPU free  RAM free   Disk free
    srv4-uswest1cdevc.dev.yelpcorp.com          0.07      30691.00   97733.00
    srv4-uswest1adevc.dev.yelpcorp.com          0.80      38941.00   107972.00
    srv1-uswest1cdevc.dev.yelpcorp.com          1.55      1409.00    256523.00
    srv2-uswest1adevc.dev.yelpcorp.com          6.87      10350.00   267787.00
    srv3-uswest1adevc.dev.yelpcorp.com          9.34      13749.00   269835.00
    srv2-uswest1cdevc.dev.yelpcorp.com          9.40      6752.00    271883.00
    srv3-uswest1cdevc.dev.yelpcorp.com          10.50     9871.00    277003.00
    srv1-uswest1adevc.dev.yelpcorp.com          12.50     15930.00   284171.00
    10-40-10-35-uswest1adevc.dev.yelpcorp.com   16.16     42503.00   240304.00
    10-40-10-243-uswest1adevc.dev.yelpcorp.com  16.57     40749.00   238256.00
    10-40-10-100-uswest1adevc.dev.yelpcorp.com  21.20     87187.00   264880.00
    10-40-10-80-uswest1adevc.dev.yelpcorp.com   27.27     69487.00   246448.00
    10-40-10-163-uswest1adevc.dev.yelpcorp.com  28.70     147987.00  276144.00
    10-40-11-114-uswest1adevc.dev.yelpcorp.com  29.30     77751.00   253616.00
    10-40-11-167-uswest1adevc.dev.yelpcorp.com  33.80     112487.00  263856.00
    10-40-10-198-uswest1adevc.dev.yelpcorp.com  36.70     135959.00  273072.00
    10-40-10-234-uswest1adevc.dev.yelpcorp.com  37.80     141687.00  277168.00
```